src/Nethermind/Nethermind.State/PartialStorageProviderBase.cs

```diff
@@ -0,0 +1,230 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Core.Collections;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Resettables;
+using Nethermind.Logging;
+using Nethermind.Trie.Pruning;
+
+namespace Nethermind.State
+{
+    public abstract class PartialStorageProviderBase
+    {
+        protected readonly ResettableDictionary<StorageCell, StackList<int>> _intraBlockCache = new();
+
+        /// <summary>
+        /// EIP-1283
+        /// </summary>
+        protected readonly ResettableDictionary<StorageCell, byte[]> _originalValues = new();
+
+        protected readonly ResettableHashSet<StorageCell> _committedThisRound = new();
+
+        protected readonly ILogger _logger;
+        protected readonly ILogManager _logManager;
+
+        protected readonly ResettableDictionary<Address, StorageTree> _storages = new();
```

Comment:
Is this needed for transient storage? Consider moving to `PersistentStorageProvider`